src/roi/Roi.ts

```diff
@@ -91,4 +138,584 @@ export class Roi {
   public getBorderPoints(options?: GetBorderPointsOptions): Array<Point> {
     return getBorderPoints(this, options);
   }
+
+  _computeBorderIDs(): { ids: number[]; lengths: number[] } {
+    let borders = getBorders(this);
+    this.#computed.borderIDs = borders.ids;
+    this.#computed.borderLengths = borders.lengths;
+    return borders;
+  }
+
+  /**
+   * Return an array of ROIs IDs that are included in the current ROI.
+   * This will be useful to know if there are some holes in the ROI.
+   */
+  get internalIDs() {
+    return this.#getComputed('internalIDs', () => {
+      return getInternalIDs(this);
+    });
+    // if (!this.#computed.internalIDs) {
+    //   this.#computed.internalIDs = getInternalIDs(this);
+    // }
+    // return this.#computed.internalIDs;
+  }
+
+  /**
+   * Return an array of ROIs IDs that touch the current ROI.
+   */
+  get externalIDs(): number[] {
+    // return this.#getComputed('externalIDs', () => {
+    //   return this.getExternalIDs().borders;
+    // });
+    if (this.#computed.externalIDs) {
+      return this.#computed.externalIDs;
+    }
+    this.getExternalIDs();
+    return this.#computed.externalIDs || [];
+  }
+
+  private get externalLengths() {
+    // return this.#getComputed('externalLengths', () => {
+    //   return this.getExternalIDs().lengths;
+    // });
+    if (!this.#computed.externalLengths) {
+      this.getExternalIDs();
+      return this.#computed.externalLengths;
+    }
+    return this.#computed.externalLengths;
+  }
+
+  get perimeterInfo() {
+    if (!this.#computed.perimeterInfo) {
+      this.#computed.perimeterInfo = getPerimeterInfo(this);
+    }
+    return this.#computed.perimeterInfo;
+  }
+
+  /**
+   * Perimeter of the ROI.
+   * The perimeter is calculated using the sum of all the external borders of the ROI to which we subtract
+   * (2 - √2) * the number of pixels that have 2 external borders
+   * 2 * (2 - √2) * the number of pixels that have 3 external borders
+   */
+  get perimeter() {
+    let info = this.perimeterInfo;
+    let delta = 2 - Math.sqrt(2);
+    return (
+      info.one +
+      info.two * 2 +
+      info.three * 3 +
+      info.four * 4 -
+      delta * (info.two + info.three * 2 + info.four)
+    );
+  }
+
+  get points() {
+    return this.#getComputed('points', () => {
+      if (!this.#computed.points) {
```

Comment:
The functions that get passed to `#getComputed` should unconditionally compute the property and return it. It should not care about the `#computed` property.